unison-core/src/Unison/DataDeclaration.hs

```diff
@@ -76,6 +77,21 @@ declDependencies = either (dependencies . toDataDecl) dependencies
 labeledDeclDependencies :: (Ord v) => Decl v a -> Set LD.LabeledDependency
 labeledDeclDependencies = Set.map LD.TypeReference . declDependencies
 
+-- | Compute the dependencies of a data declaration,
+-- including the type itself and references for each of its constructors.
+labeledDeclDependenciesIncludingSelf :: (Ord v) => Reference.TypeReference -> Decl v a -> Set LD.LabeledDependency
```

Comment:
The other dependency methods  don't include the constructors themselves in the "dependencies", only the arguments of those constructors. But for pretty printing we need the constructor refs too.

It's not ideal  that this takes the TypeReference as an argument, but I'm not sure there's a better solution, it'd be needlessly costly to re-compute it inside the function.